source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
Meanwhile, I filed a [spec bug](https://www.khronos.org/bugzilla/show_bug.cgi?id=1471) for this and will ensure it's addressed pronto by the SPIR-V working group.